### DISCUSSION TITLE
Full database rebuild - incorrect counts

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1347] [Fixed Issues]
RPi 20230609 - local database

A full database rebuild seems to be counting Albums and populating the Library differently than previous versions.  I have 336 albums (as counted on previous versions) but after updating to 20230609 and carrying out a full database rebuild I now  apparently have 607!  This version seems to be counting artists (?) rather than individual albums.  After the rebuild, in Library / Albums, I am seeing single entries for some albums and multiple entries for others - Abba Gold has 9 entries, Elton John Greatest Hits 1970 - 2002 has 19 !

I was initially seeing this on a Shared Data installation but it is also doing exactly the same on a new local build.

EDIT 1

The database update process also tries to  create thumbnail files in the wrong place, eg:

journalctl -xe shows the following error:

Jun 10 18:12:44 rAudio-1 nginx[385]: 2023/06/10 18:12:44 [error] 385#385: *217 open() "/srv/http/mnt/MPD/NAS/NAS_Music/Alfie Boe/thumb.jpg" failed (2: No such file or directory), client: 192.168.86.53, server: localhost, request:
"GET /%2Fmnt%2FMPD%2FNAS%2FNAS_Music%2FAlfie%20Boe%2Fthumb.jpg?v=1686417165 HTTP/1.1", host: "raudio-1", referrer: "http://raudio-1/"

however:

rAudio-1:~  #  ls -lrt "/srv/http/mnt/MPD/NAS/NAS_Music/Alfie Boe/thumb.jpg"        
ls: cannot access '/srv/http/mnt/MPD/NAS/NAS_Music/Alfie Boe/thumb.jpg': No such file or directory

It should be:

rAudio-1:~  #  ls -lrt "/srv/http/mnt/MPD/NAS/NAS_Music/Alfie Boe/Bring Him Home"
total 241800
-rwxrwxrwx+ 1 root 100 22564876 Dec  1  2021 '01 Bring Him Home.m4a'
-rwxrwxrwx+ 1 root 100 17450918 Dec  1  2021 '02 Pure Imagination.m4a'
-rwxrwxrwx+ 1 root 100 29964913 Dec  1  2021 '03 Come What May.m4a'
-rwxrwxrwx+ 1 root 100 14067853 Dec  1  2021 '04 Hushabye Mountain.m4a'
-rwxrwxrwx+ 1 root 100 29922570 Dec  1  2021 '05 As If We Never Said Goodbye.m4a'
-rwxrwxrwx+ 1 root 100 21177384 Dec  1  2021 '06 On The Street Where You Live.m4a'
-rwxrwxrwx+ 1 root 100 19004848 Dec  1  2021 '07 Some Enchanted Evening.m4a'
-rwxrwxrwx+ 1 root 100 18280963 Dec  1  2021 '08 The Impossible Dream.m4a'
-rwxrwxrwx+ 1 root 100 16605588 Dec  1  2021 '09 If I Loved You.m4a'
-rwxrwxrwx+ 1 root 100 23705102 Dec  1  2021 "10 Tell Me It's Not True.m4a"
-rwxrwxrwx+ 1 root 100 15022683 Dec  1  2021 '11 If I Ruled The World.m4a'
-rwxrwxrwx+ 1 root 100 19738339 Dec  1  2021 '12 We Have All The Time In The World.m4a'
-rwxrwxrwx+ 1 root 100    37340 Dec  1  2021  folder.jpg
-rwxrwxrwx+ 1 root 100    15226 Dec  1  2021  coverart.jpg
-rwxrwxrwx+ 1 root 100     3970 Dec  1  2021  thumb.jpg
-rwxrwxrwx+ 1 root 100     8111 Dec  1  2021  AlbumArtSmall.jpg

and as you can see the thumb.jpg file already exists in this folder as it has been in use with my rAudio setup for quite a while..

I seem to remember this being a problem quite a while ago but was fixed at that time.

Chris

**Comment by Editpo** [comment]
I am also seeing multiple entries for some albums.

**Reply by Sagnasty** [reply]
Had the same issue after updating to 20230609

The library update will now give you multiple instances of the same album if the album tracks have different years for that specific album.

This is easier to detect if you look at the library view for multiple instances of the same album art work.

Jim

**Reply by Sagnasty** [reply]
Edit - This is easier to detect if you look at the library / album view for multiple instances of the same album art work.

**Comment by Editpo** [comment]
@Sagnasty yes that has fixed multiple albums for me. Thanks.
Still got a few anomalies in the library which I need to investigate.

**Comment by ceejayemm** [comment]
@Editpo 

What did you do to fix this ?  I haven't seen anything which suggests there is a fix.

Chris

**Reply by kw-sudo** [reply]
[master-kw]
> my database contains three folders: Album, Sampler (mostly VA) and Singles. The latter do not contain an album tag but are still displayed in the album overview (without cover, now ~3k 'albums' vs 825 real ones). In 0603 it was fine.

I did another clean install from the scratch (i20230521 + addon 20230609, then pacman -Syu and backup config 20230609). Still the same error in the library. Then ('against the guidelines') imported the older config backup 20230603, the listing was correct again.

Oliver

**Reply by Editpo** [reply]
@ceejayemm
I did what sagnasty suggested
"The library update will now give you multiple instances of the same album if the album tracks have different years for that specific album."

I had a look at some albums which appeared multiple times under the tag editor and found some tracks had different years on them. I deleted the year, did a library update and the albums now showed as a single album

**Comment by kw-sudo** [comment]
I think I found the 'nasty troublemaker'. Checked the library settings, the new option '_sort by artist > year_' was off. Turning on, the muddle begins, turning off, cleared the situation again.

**Reply by kw-sudo** [reply]
Sorry, folks, I have to row back. Once updated the library and the visualization is not correct again. I think I will stay with i202130521 for that long.

Oliver

**Comment by ceejayemm** [comment]
A full database rebuild, with 'Sort by Album > Year' enabled is counting the number of different years found in the tracks of the album, which might be right for some people, but perhaps should really reflect the release date of the entire album rather than the dates of individual tracks.  It is a particular problem for compilation type albums , so from my example above there are 9 different years amongst the tracks of Abba Gold and 19 for Elton John Greatest Hits 1970 - 2002. Exactly the number of images produced once the full database rebuild with this option enabled completes.  The answer seems to be to turn this option OFF to get an accurate count of albums.

Chris

**Reply by kw-sudo** [reply]
Chris, I can confirm that. I just 'dared' to upgrade my i20230521 + pacman -Syu version to 20230609 and leave the option OFF.

Works! :-)

PS: Originally, I was at least on the right track as far as the error pattern was concerned.

Oliver

**Reply by kw-sudo** [reply]
Return command. Just added a new album, updated the library...'same procedure as last year...'

Oliver

**Reply by kw-sudo** [reply]
Another step: unchecked _"album -> sort by artist"_ (sorted quasi by album name), listing is okay. Checked this box again, listing error.

**Comment by ceejayemm** [comment]
With option 'Album -> Sort by Artist ' enabled, a full rebuild recreates the Library as I would expect.

Chris

**Reply by kw-sudo** [reply]
@ceejayemm

just reached this step... Result :-(


![library](https://github.com/rern/rAudio/assets/104136223/358017b6-fb73-4d44-8ee4-5ea4e8cc25b6)

**Comment by ceejayemm** [comment]
@master-kw 
Was this the first time you have built the Library database (for this installation) ? If so, I think (?) that the latest versions of rAudio default to the new 'Sort by Artist > Year' even if you have not enabled it.  Go to Library options (from the Cog Wheel  (I wish there was a better way to describe this) and under Library Options enable 'Albums - Sort by Artist'. You might also want to enable / disable  'Sort by Artist > Year' just to be sure its not active (!)  Then do a full database rebuild. 

Chris

**Reply by kw-sudo** [reply]
@ceejayemm 
Yes, I did, once again, a clean install from the scratch (fortunately, I struck at the last SD card offer at Amazon, so I can switch ;-)).

In the context of the other errors that are circulating here, I could finish all the the 'basic settings' (player, network, system, etc.). When I came to the library settings, all 'album/player boxes' are unchecked. I only selected 'Albums - Sort by Artist' and did a full database rebuild. 

The result is the thing above.
